File: libgdx_sokoban/core/src/com/nopalsoft/sokoban/scene2d/TouchPadControlsTable.java
package com.nopalsoft.sokoban.scene2d;

import com.badlogic.gdx.scenes.scene2d.InputEvent;
import com.badlogic.gdx.scenes.scene2d.ui.Button;
import com.badlogic.gdx.scenes.scene2d.ui.Table;
import com.badlogic.gdx.scenes.scene2d.utils.ClickListener;
import com.nopalsoft.sokoban.Assets;
import com.nopalsoft.sokoban.game.GameScreen;

public class TouchPadControlsTable extends Table {

    GameScreen gameScreen;

    Button buttonUp, buttonDown, buttonLeft, buttonRight;

    public TouchPadControlsTable(GameScreen oScreen) {
        gameScreen = oScreen;

        getColor().a = .4f;
        init();

        int buttonSize = 75;
        defaults().size(buttonSize);

        add(buttonUp).colspan(2).center();
        row();
        add(buttonLeft).left();
        add(buttonRight).right().padLeft(buttonSize / 1.15f);
        row();
        add(buttonDown).colspan(2).center();
        pack();
    }

    private void init() {
        buttonUp = new Button(Assets.buttonUpDrawable, Assets.buttonUpPressedDrawable);
        buttonUp.addListener(new ClickListener() {
            @Override
            public void clicked(InputEvent event, float x, float y) {
                gameScreen.up();
            }
        });

        buttonDown = new Button(Assets.buttonDownDrawable, Assets.buttonDownPressedDrawable);
        buttonDown.addListener(new ClickListener() {
            @Override
            public void clicked(InputEvent event, float x, float y) {
                gameScreen.down();
            }
        });

        buttonLeft = new Button(Assets.buttonLeftDrawable, Assets.buttonLeftPressedDrawable);
        buttonLeft.addListener(new ClickListener() {
            @Override
            public void clicked(InputEvent event, float x, float y) {
                gameScreen.left();
            }
        });

        buttonRight = new Button(Assets.buttonRightDrawable, Assets.buttonRightPressedDrawable);
        buttonRight.addListener(new ClickListener() {
            @Override
            public void clicked(InputEvent event, float x, float y) {
                gameScreen.right();
            }
        });
    }
}

File: libgdx_sokoban/core/src/com/nopalsoft/sokoban/scene2d/TouchPadControlsTable.kt
package com.nopalsoft.sokoban.scene2d

import com.badlogic.gdx.scenes.scene2d.InputEvent
import com.badlogic.gdx.scenes.scene2d.ui.Button
import com.badlogic.gdx.scenes.scene2d.ui.Table
import com.badlogic.gdx.scenes.scene2d.utils.ClickListener
import com.nopalsoft.sokoban.Assets
import com.nopalsoft.sokoban.game.GameScreen

class TouchPadControlsTable(var gameScreen: GameScreen) : Table() {
    var buttonUp: Button? = null
    var buttonDown: Button? = null
    var buttonLeft: Button? = null
    var buttonRight: Button? = null

    init {
        color.a = .4f
        init()

        val buttonSize = 75
        defaults().size(buttonSize.toFloat())

        add(buttonUp).colspan(2).center()
        row()
        add(buttonLeft).left()
        add(buttonRight).right().padLeft(buttonSize / 1.15f)
        row()
        add(buttonDown).colspan(2).center()
        pack()
    }

    private fun init() {
        buttonUp = Button(Assets.buttonUpDrawable, Assets.buttonUpPressedDrawable)
        buttonUp!!.addListener(object : ClickListener() {
            override fun clicked(event: InputEvent, x: Float, y: Float) {
                gameScreen.up()
            }
        })

        buttonDown = Button(Assets.buttonDownDrawable, Assets.buttonDownPressedDrawable)
        buttonDown!!.addListener(object : ClickListener() {
            override fun clicked(event: InputEvent, x: Float, y: Float) {
                gameScreen.down()
            }
        })

        buttonLeft = Button(Assets.buttonLeftDrawable, Assets.buttonLeftPressedDrawable)
        buttonLeft!!.addListener(object : ClickListener() {
            override fun clicked(event: InputEvent, x: Float, y: Float) {
                gameScreen.left()
            }
        })

        buttonRight = Button(Assets.buttonRightDrawable, Assets.buttonRightPressedDrawable)
        buttonRight!!.addListener(object : ClickListener() {
            override fun clicked(event: InputEvent, x: Float, y: Float) {
                gameScreen.right()
            }
        })
    }
}
